addons/web/models/models.py

```diff
@@ -141,7 +141,11 @@ def cleanup(vals: Dict) -> Dict:
                         for index, co_record in enumerate(co_records)
                     }
                     for values in values_list:
-                        values[field_name] = sorted(values[field_name], key=order_key.__getitem__)
+                        values[field_name] = sorted(
+                            # this hack filters out inaccessible corecords
+                            [id_ for id_ in values[field_name] if id_ in order_key],
```

Comment:
```suggestion
                            values.keys() & order_key,
```
also works

Reply:
the .keys() need to be on the order_key dict, and we need the values[field_name] anyway but yes it's cleaner